src/main/java/com/gradle/develocity/bamboo/MavenBuildScanInjector.java

```diff
@@ -90,36 +94,62 @@ private void inject(BuildContext buildContext, Collection<RuntimeTaskDefinition>
             return;
         }
 
+        MavenExtensions existingMavenExtensions = getExistingMavenExtensions(buildContext);
+
         Classpath classpath = new Classpath();
-        classpath.add(mavenEmbeddedResources.copy(MavenEmbeddedResources.Resource.DEVELOCITY_EXTENSION));
-        if (config.injectCcudExtension) {
+        List<SystemProperty> systemProperties = new ArrayList<>();
+        if (!existingMavenExtensions.hasExtension(DEVELOCITY_EXTENSION_MAVEN_COORDINATES)) {
```

Comment:
Is there a case/need to enforce the DV URL configured in Bamboo when the extension is already set?

Reply:
We didn't have it as a feature yet, but it can be introduced.
I'm just not sure if it should be in this PR, as it kind of a separate thing.

Reply:
Yeah, separate is better